pkg/addon/templateagent/template_agent.go

```diff
@@ -96,6 +97,11 @@ func (a *CRDTemplateAgentAddon) Manifests(
 	cluster *clusterv1.ManagedCluster,
 	addon *addonapiv1alpha1.ManagedClusterAddOn) ([]runtime.Object, error) {
 
+	if !meta.IsStatusConditionTrue(addon.Status.Conditions, addonapiv1alpha1.ManagedClusterAddOnConditionConfigured) {
```

Comment:
if we have addon-framework check the conds, do we still need this?

Reply:
Currently, the code to "check the conds" is added at the beginning of each implementation of `Manifests()`, so both template and addon-framework have this piece of code.

Reply:
will try to find a better place to add the check.